homeassistant/components/nfandroidtv/config_flow.py

```diff
@@ -0,0 +1,82 @@
+"""Config flow for NFAndroidTV integration."""
+from __future__ import annotations
+
+from asyncio.exceptions import TimeoutError
+import logging
+
+from aiohttp.client_exceptions import ClientConnectorError
+from async_timeout import timeout
+from httpcore import ConnectError
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_HOST, CONF_NAME
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import DEFAULT_NAME, DEFAULT_TIMEOUT, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class NFAndroidTVFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for NFAndroidTV."""
+
+    async def async_step_user(self, user_input=None) -> FlowResult:
+        """Handle a flow initiated by the user."""
+        errors = {}
+
+        if user_input is not None:
+            host = user_input[CONF_HOST]
+            name = user_input[CONF_NAME]
+
+            error = await self._async_try_connect(host)
+            if error is None:
+                await self.async_set_unique_id(f"{host}_{DOMAIN}")
```

Comment:
This can be corrected with getting the mac address using dhcp discovery which I plan on doing. I would need any hostname prefixes that android tv devices use since I don't have one, just fire tv.

Reply:
I think AndroidTV uses `_androidtvremote._tcp` as service name.

Reply:
Any suggestions on a unique id? I'm not sure how to use config entry id as the last resort.

Reply:
We don't have to post or prefix it with the DOMAIN, it is already unique within the namespace of the integration domain.

This is not the entity registry and therefore does not have the same unique ID requirements as entities have. So the host can be used (given that a single used can only be configured once).